srcs/raycast.c

```diff
@@ -0,0 +1,170 @@
+#include "cub.h"
+
+void	initialize_map_pos(t_cub *cub)
+{
+	cub->time = 0;
+	cub->old_time = 0;
+}
+
+/*
+	SET_RAY_DIR:
+	- Add to the direction vector, the projection plane vector and
+	multiply it by the x-coordinate in camera space
+	- Vectors have both an x and y component, so do this for both dimensions
+*/
+void	set_ray_dir(t_cub *cub)
```

Comment:
recap what we discussed: move memory allocation to init_cub from all these functions (set_ray_dir, set_delta_dir, initialize_raycasting + use calloc_or_exit instead
and change name from initialize_raycasting to set_raycasting_vars or something similar

Reply:
Added a raycasting folder now with raycast.c and draw_line.c

Additionally, I renamed the draw_pixel function to set_pixel, as the function only sets the location where the pixel should be drawn, does not actually draw it.